src/mongo_odm/query_builder.hpp

```diff
@@ -120,15 +481,53 @@ class NotExpr {
     }
 
    private:
-    const ComparisonExpr<Base, T> _expr;
+    const Expr _expr;
 };
 
+/**
+ * This represents an expression without a field name, for instance used inside $elemMatch queries
+ * for scalar arrays.
+ * The class wraps an existing query expression, and omits the field name when converting to BSON.
+ */
+template <typename Expr>
+class FreeExpr {
+   public:
+    constexpr FreeExpr(Expr expr) : expr(expr) {
+    }
+
+    /**
+     * Appends this expression to a BSON builder, in the form {$op: val} (without a field name)
+     * @param builder A BSON core builder.
+     * @param wrap    Whether to contain this expression inside its own document.
+     */
+    void append_to_bson(bsoncxx::builder::core &builder, bool wrap = false) const {
+        if (wrap) {
+            builder.open_document();
+        }
+        // Append expression without name
+        expr.append_to_bson(builder, false, true);
+
+        if (wrap) {
+            builder.close_document();
+        }
+    }
+
+    // This should never be top-level, and thus does not have a conversion operator directly to
+    // BSON.
+
+   private:
+    const Expr expr;
+};
+
+/**
+ * This represents a list of expressions.
+ */
 template <typename Head, typename Tail>
 class ExpressionList {
```

Comment:
Suggestion, add a non-type template param of an enum to distinguish list type. e.g.
```c++
enum struct expression_list_type {
    k_update,
    k_query,
    k_sort,
    //etc.
}
```